src/server/zset_family.cc

```diff
@@ -2840,54 +2820,134 @@ void SortIfNeeded(GeoArray* ga, Sorting sorting, uint64_t count) {
   }
 }
 
-void GeoSearchGeneric(ConnectionContext* cntx, const GeoShape& shape_ref, string_view key,
-                      const GeoSearchOpts& geo_ops) {
-  // query
+// write_mode: Command in WRITE mode
+//             True: Using transaction->Execute()
+//             False:  Using transaction->ScheduleSingleHopT()
+// store_key:  (Valid if geo_ops.store==GeoStoreType::kNoStore) Store result to store_key
+void GeoSearchStoreGeneric(ConnectionContext* cntx, const GeoShape& shape_ref, string_view key,
```

Comment:
this function is always called on already scheduled transactions. That means it never should call ScheduleSingleHop because this function is only intended to be used for non-scheduled transactions